lib/ansible/modules/net_tools/basics/get_url.py

```diff
@@ -614,6 +614,18 @@ def main():
                              result['checksum_src'] != result['checksum_dest'])
         module.exit_json(msg=info.get('msg', ''), **result)
 
+    # If a checksum was provided, ensure that the temporary file matches this checksum
+    # before moving it to the destination.
+    if checksum != '':
+        tmpsrc_checksum = module.digest_from_file(tmpsrc, algorithm)
+
+        if checksum != tmpsrc_checksum:
+            os.remove(tmpsrc)
+            module.fail_json(msg="The checksum for %s did not match %s; it was %s." % (
+                tmpsrc, checksum, tmpsrc_checksum
+            ), **result)
```

Comment:
I'm not a huge fan of this formatting, it's actually rather confusing and doesn't read well.  Our line length limit is 160 for this project, so you can mostly just inline it.  The original line is more readable than what this became.

```suggestion
            module.fail_json(msg="The checksum for %s did not match %s; it was %s." % (tmpsrc, checksum, tmpsrc_checksum), **result)
```

Reply:
Fixed.